scripts/check_imports_and_dependencies.py

```diff
@@ -0,0 +1,271 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Check aea dependencies."""
+import copy
+import dis
+import importlib
+import re
+import sys
+from collections import defaultdict
+from functools import wraps
+from pathlib import Path
+from typing import Any, Callable, Dict, Generator, List, Set, Tuple, Union
+
+from pip._internal.commands.show import search_packages_info  # type: ignore
+
+
+AEA_ROOT_DIR = Path(__file__).parent.parent
+
+sys.path.append(str(AEA_ROOT_DIR))
+
+from aea.crypto.registries import (  # noqa # pylint: disable=wrong-import-position
+    crypto_registry,
+)
+
+
+DEP_NAME_RE = re.compile(r"(^[^=><\[]+)", re.I)  # type: ignore
+
+
+def list_decorator(fn: Callable) -> Callable:
+    """Wraps generator to return list."""
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs) -> List[Any]:
+        return list(fn(*args, **kwargs))
+
+    return wrapper
+
+
+class DependenciesTool:
+    """Tool to work with setup.py dependencies."""
+
+    @staticmethod
+    def get_package_files(package_name: str) -> List[str]:
+        """Get package files list."""
+        packages_info = list(search_packages_info([package_name]))
+        if len(packages_info) == 0:
+            raise Exception(f"package {package_name} not found")
+        files = packages_info[0]["files"]
+        location = packages_info[0]["location"]
+        return [str(Path(location) / i) for i in files]
+
+    @staticmethod
+    def clean_dependency_name(dependecy_specification: str) -> str:
+        """Get dependency name from dependency specification."""
+        match = DEP_NAME_RE.match(dependecy_specification)
+        if not match:
+            raise ValueError(f"Bad dependency specification: {dependecy_specification}")
+        return match.groups()[0]
+
+
+class ImportsTool:
+    """Tool to work with 3rd part imports in source code."""
+
+    @staticmethod
+    def get_imports_for_file(pyfile: Union[str, Path]) -> List[str]:
+        """Get all imported modules for python source file."""
+        with open(pyfile, "r") as f:
+            statements = f.read()
+        instructions = dis.get_instructions(statements)  # type: ignore
+        imports = [i for i in instructions if "IMPORT" in i.opname]
+
+        grouped: Dict[str, List[str]] = defaultdict(list)
+        for instr in imports:
+            grouped[instr.opname].append(instr.argval)
+
+        return grouped["IMPORT_NAME"]
+
+    @staticmethod
+    def get_module_file(module_name: str) -> str:
+        """Get module source file name."""
+        try:
+            mod = importlib.import_module(module_name)
+            return getattr(mod, "__file__", "")
+        except (AttributeError, ModuleNotFoundError):
+            return ""
```

Comment:
`__file__` is optional, so this might fail in some cases, right?
https://docs.python.org/3/reference/import.html#__file__

Reply:
maybe not, since `load_module` sets `__file__`:

https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.load_module

Reply:
anyway I need __file__ to find out what is the package.

Reply:
yes; but we could find it by looking into `site-packages/python3.X` in case `__file__` is not present. Anyway, what I said is overkill, this should always work.